### PR TITLE
sends emails when device is lent and returned

### LINKED ISSUE #123: Send email to users when device is lent or returned

### PR DESCRIPTION
**Pull request by LizzAlice**
fixes #123

ATTENTION: for emails to be sent, the according MailTemplates have to be created!

### CODE REVIEW

File: mail/forms.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 from mail.models import MailTemplate
 from devices.forms import get_emailrecipientlist
-
+import pdb
```
Comment (octomike) [inline mail/forms.py]: :eyes:

File: Lagerregal/base_settings.py [hunk 2]
```diff
@@ -135,9 +135,9 @@
 
 SITE_NAME = "Lagerregal"
 
-MESSAGE_TAGS = {
-    messages.ERROR: 'alert',
-}
+# MESSAGE_TAGS = {
+#     messages.ERROR: 'alert',
+# }
```
Comment (octomike) [inline Lagerregal/base_settings.py]: what's happening here?

File: Lagerregal/template_development.py [hunk 3]
```diff
@@ -51,3 +51,9 @@
 INTERNAL_IPS = [
     "127.0.0.1",
 ]
+
+#comment out variables and start server with
+#python -m smtpd -n -c DebuggingServer localhost:1025
+#to print emails to console
+# EMAIL_HOST = 'localhost'
+# EMAIL_PORT = 1025
```
Comment (octomike) [inline Lagerregal/template_development.py]: Can you put that into another PR, please? :)

File: mail/ajax.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+from mail.models import MailTemplate
+
+class LoadChoices(View):
+    def post(self, request, *args, **kwargs):
+        form = MailTemplateForm(request.POST, initial={'department': request.POST["department"] })
+        department = request.POST["department"]
+        used = MailTemplate.objects.values_list('usage', flat=True).filter(department = department)
+        valid = {str(x):y for (x,y) in usages.items() if not any(z in x for z in used)}
+        json_valid = valid
```
Comment (octomike) [inline mail/ajax.py]: not used

File: mail/ajax.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        json_valid = {}
+        for choice in valid:
+            json_valid[choice] = unicode(usages[choice])
```
Comment (octomike) [inline mail/ajax.py]: this could use a comment

File: mail/models.py [hunk 6]
```diff
@@ -25,9 +25,9 @@ class MailTemplate(models.Model):
     subject = models.CharField(_('Subject'), max_length=500)
     body = models.CharField(_('Body'), max_length=10000)
     department = models.ForeignKey(Department, null=True)
-
     usage = models.CharField(_('Usage'), choices=usages.items(), null=True, blank=True, max_length=200)
 
+
```
Comment (octomike) [inline mail/models.py]: is that PEP8 or why the newline?

File: mail/tests.py [hunk 7]
```diff
@@ -29,6 +29,7 @@ def test_template_creation(self):
 
 
 
+
```
Comment (octomike) [inline mail/tests.py]: s.a.

File: mail/views.py [hunk 8]
```diff
@@ -6,10 +6,12 @@
 from django.shortcuts import get_object_or_404
 
 from mail.models import MailTemplate, MailTemplateRecipient
-from mail.forms import MailTemplateForm
+from mail.forms import MailTemplateUpdateForm, MailTemplateForm
 from users.models import Lageruser
 from Lagerregal.utils import PaginationMixin
 
+import pdb
```
Comment (octomike) [inline mail/views.py]: :eyes:

File: templates/devices/base_form.html [hunk 9]
```diff
@@ -58,12 +58,28 @@ <h3 style="margin-top:0px">{% trans "Help" %}</h3>
...
+      $.ajax({
+        type: "POST",
+        url : "{% url "load-choices" %}",
+        data : {"department" : department},
+        success: function(data){
+          $("#id_usage").empty();
+          for(key in data){
+            $('#id_usage').append('<option value="' + key + '">' + data[key] +'</option>');
+          }
+          $("#id_usage").each(function(){
```
Comment (octomike) [inline templates/devices/base_form.html]: so `map()` after all :)

File: templates/devices/base_form.html [hunk 10]
```diff
@@ -58,12 +58,28 @@ <h3 style="margin-top:0px">{% trans "Help" %}</h3>
...
+            $(this).selectedIndex=0;
+});
```
Comment (octomike) [inline templates/devices/base_form.html]: indentation

File: mail/forms.py (hunk 1 above)
Comment (LizzAlice) [reply mail/forms.py]: done

File: Lagerregal/base_settings.py (hunk 2 above)
Comment (LizzAlice) [reply Lagerregal/base_settings.py]: That stopped the error messages from having the right color, looks like it was inserted to customize the error, but I couldn't find any reason for it, so I commented it out instead of deleting it... Should I delete it?

File: mail/ajax.py (hunk 4 above)
Comment (LizzAlice) [reply mail/ajax.py]: did you meant the form-part above that? I removed that one, but the lines that are marked here are used.

File: mail/ajax.py (hunk 5 above)
Comment (LizzAlice) [reply mail/ajax.py]: ok

File: mail/models.py (hunk 6 above)
Comment (LizzAlice) [reply mail/models.py]: fixed it

File: mail/tests.py (hunk 7 above)
Comment (LizzAlice) [reply mail/tests.py]: ok

File: mail/views.py (hunk 8 above)
Comment (LizzAlice) [reply mail/views.py]: removed
